tests/common/fixtures/advanced_reboot.py

```diff
@@ -0,0 +1,510 @@
+import ipaddress
+import itertools
+import json
+import logging
+import pytest
+import time
+
+from common.errors import RunAnsibleModuleFail
+from common.mellanox_data import is_mellanox_device as isMellanoxDevice
+from common.reboot import reboot as rebootDut
+from ptf_runner import ptf_runner
+
+logger = logging.getLogger(__name__)
+
+# Globals
+HOST_MAC_COUNT = 126
+TIME_BETWEEN_SUCCESSIVE_TEST_OPER = 420
+PTFRUNNER_QLEN = 1000
+
+class AdvancedReboot:
+    '''
+    AdvancedReboot class collects information about the current testbed. This information is used by test cases to build
```

Comment:
In general the best practice for class docstrings is to have a one-line summary followed by a more detailed description, like this:

```
AdvancedReboot is used to ...

It will collect information ... This is used for ... Etc...
```